src/Container/App/Mail/TransportFactory.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Prooph\ProophessorDo\Container\App\Mail;
+
+use Interop\Config\ConfigurationTrait;
+use Interop\Config\ProvidesDefaultOptions;
+use Interop\Config\RequiresConfig;
+use Interop\Config\RequiresMandatoryOptions;
+use Interop\Container\ContainerInterface;
+use Zend\Mail\Transport\InMemory as InMemoryTransport;
+use Zend\Mail\Transport\Smtp as SmtpTransport;
+use Zend\Mail\Transport\SmtpOptions;
+use Zend\Mail\Transport\TransportInterface;
+
+
+/**
+ * Class TransportFactory
+ *
+ * @package Prooph\ProophessorDo\Container\Model\Todo
+ * @author Michał Żukowski <michal@durooil.com>
+ */
+final class TransportFactory implements RequiresConfig, RequiresMandatoryOptions, ProvidesDefaultOptions
+{
+    use ConfigurationTrait;
+
+    /**
+     * @inheritdoc
+     */
+    public function dimensions()
+    {
+        return ['proophessor-do', 'mail'];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function mandatoryOptions()
+    {
+        return ['transport'];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function defaultOptions()
+    {
+        return ['transport' => 'in_memory'];
+    }
+
+    /**
+     * @param ContainerInterface $container
+     * @return TransportInterface
+     */
+    public function __invoke(ContainerInterface $container)
+    {
+        return $this->getTransport($container->get('config'));
+    }
+
+    private function getTransport($config)
+    {
+        $config = $this->optionsWithFallback($config);
+
+        switch ($config['transport']) {
```

Comment:
[Existing mail configuration](https://github.com/prooph/proophessor-do/blob/master/config/autoload/mail_connection.local.php.dist) is different from the one used in mentioned Factory.